### PR TITLE
add local fonts and update themes

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

Google Fonts stores user IP addresses, so any apps using Google Fonts aren't actually GDPR compliant. 

This PR stores the required font files in our static directory, creates a new `LocalFont` class in `fonts.py` and updates our default and base themes to use that instead of `GoogleFont`. 

Q: What should we do about our other themes? We're not GDPR compliant if any of those other themes use Google Fonts. 

Closes: #7968

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/themes/utils/fonts.py [hunk 1]
```diff
@@ -52,5 +53,30 @@ def __init__(self, name: str, weights: Iterable[int] = (400, 600)):
...
+        return {"url": url, "css": None}
+
+class LocalFont(Font):
+    def __init__(self, name: str, weights: Iterable[int] = (400, 700)):
+        super().__init__(name)
+        self.weights = weights
+
+    def stylesheet(self) -> dict:
+        css_template = """
+        @font-face {{
```
Comment (freddyaboulton) [inline gradio/themes/utils/fonts.py:67]: nit: might want to de-dent with `import textwrap; textwrap.dedent`

File: gradio/themes/base.py [hunk 2]
```diff
@@ -462,26 +468,32 @@ def expand_shortcut(shortcut, mode="color", prefix=None):
...
 
         self._stylesheets = []
+        self._font_css = []
         for font in self._font + self._font_mono:
             font_stylesheet = font.stylesheet()
-            if font_stylesheet:
+            if isinstance(font_stylesheet, str):
                 self._stylesheets.append(font_stylesheet)
+            elif isinstance(font_stylesheet, dict):
+                if font_stylesheet["url"]:
```
Comment (freddyaboulton) [inline gradio/themes/base.py:493]: It's one or the other, no?

**Review by freddyaboulton — APPROVED** [review]
Wow very cool @hannahblair !

**Comment by abidlabs** [comment]
Nice @hannahblair this is great!

> Q: What should we do about our other themes? We're not GDPR compliant if any of those other themes use Google Fonts.

Could we store the fonts for all of our built-in themes as local fonts (maybe that is only IBMPlex)? For themes that are not built-in, the user needs to make a request to fetch the theme anyways, so there is less expectation of being offline

**Comment by freddyaboulton** [comment]
Fixed the issue @hannahblair - will merge once CI passes!

**Pull request merged**